include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp

```diff
@@ -0,0 +1,2889 @@
+            const std::string& type_name);
+
+    /*************** Auxiliary public methods ***************************/
+
+    /**
+     * @brief Calculate the MD5 hash of the provided name.
+     *
+     * @param[in] name String which hash is calculated.
+     * @return const NameHash Hash of the given string.
+     */
+    RTPS_DllAPI static const NameHash name_hash(
+            const std::string& name);
+
+private:
+
+    friend class TypeObjectRegistry;
+
+    // Class with only static methods
+    TypeObjectUtils() = delete;
+    ~TypeObjectUtils() = delete;
+
+protected:
+
+    /*************** Auxiliary methods ***************************/
+
+    /**
+     * @brief Set the try construct behavior in a given MemberFlag
+     *
+     * @param[in out] member_flag Bitmask to be set.
+     * @param[in] try_construct_kind TryConstructKind.
+     */
+    static void set_try_construct_behavior(
+            MemberFlag& member_flag,
+            TryConstructKind try_construct_kind);
+
+    /**
+     * @brief Set the TypeFlag object.
+     *
+     * @param[in out] type_flag Bitmask to be set.
+     * @param[in] extensibility_kind ExtensibilityKind
+     * @param[in] nested nested annotation value.
+     * @param[in] autoid_hash autoid annotation has HASH value.
+     */
+    static void set_type_flag(
+            TypeFlag& type_flag,
+            ExtensibilityKind extensibility_kind,
+            bool nested,
+            bool autoid_hash);
+
+    /**
+     * @brief Set the extensibility kind in a given TypeFlag.
+     *
+     * @param[in out] type_flag Bitmask to be set.
+     * @param[in] extensibility_kind ExtensibilityKind.
+     */
+    static void set_extensibility_kind(
+            TypeFlag& type_flag,
+            ExtensibilityKind extensibility_kind);
+
+    /**
+     * @brief Check if a given TypeIdentifier is fully-descriptive.
+     *        XTypes v1.3 Clause 7.3.4.6.1 Fully-descriptive TypeIdentifiers
+     *        Some TypeIdentifiers do not involve computing the Hash of any TypeObject. These are called
+     *        Fully-descriptive TypeIdentifiers because they fully describe the Type. These are:
+     *        - The TypeIdentifiers for Primitive and String types.
+     *        - The TypeIdentifiers of plain collections where the element (and key) TypeIdentifier is a fully
+     *          descriptive TypeIdentifier. They are recognized by the contained PlainCollectionHeader having
+     *          EquivalenceKind set to EK_BOTH.
+     *
+     * @param[in] type_identifier TypeIdentifier to check.
+     * @return true if the given TypeIdentifier is fully-descriptive. false otherwise.
+     */
+    static bool is_fully_descriptive_type_identifier(
+            const TypeIdentifier& type_identifier);
+
+    /**
+     * @brief Check if a given TypeIdentifier is direct hash.
+     *        XTypes v1.3 Clause 7.3.4.6.3
+     *        These are HASH TypeIdentifiers with discriminator EK_MINIMAL, EK_COMPLETE or TI_STRONG_COMPONENT.
```

Comment:
```suggestion
     *        These are HASH TypeIdentifiers with discriminator EK_MINIMAL, EK_COMPLETE or TI_STRONGLY_CONNECTED_COMPONENT.
```

Reply:
@adriancampo should this comment be moved to #3974?

Reply:
done